src/runtime/logger.js

```diff
@@ -20,14 +20,20 @@ function getLogger (params) {
 
   const transports = [
     new winston.transports.Console({ format }),
-    new winston.transports.File({ filename: 'stdout', format })
   ]
 
-  if (params.testOutputsPath) {
+  if (process.title !== 'browser') {
     transports.push(new winston.transports.File({
       format,
-      filename: path.join(params.testOutputsPath, 'run.out')
+      filename: 'stdout'
     }))
+
+    if (!params.testOutputsPath) {
+      transports.push(new winston.transports.File({
+        format,
+        filename: path.join(params.testOutputsPath, 'run.out')
+      }))
+    }
   }
 
   return winston.createLogger({
```

Comment:
(creating a thread)

You listed a lot of `fallback` libraries, do you know what purpose they solve? (for example the path library might "just" be required by the winston logger).

@achingbrain shared an interesting option: It might be enough to use the `browser` field to shim the features that are not isomorphic (the loggers creation and the env retrieval): https://github.com/defunctzombie/package-browser-field-spec#replace-specific-files---advanced

Reply:
They solve the purpose of being able to run the testground plan without runtime errors, so that collection get assembled in a trial-error approach. I'll look into the browser field, haven't used that before, so if you could give some practical example of how you see that that would be great. Either way I'll give it a look.

Reply:
Ok got'cha, you shared it in the other thread already :) 
(https://github.com/testground/sdk-js/pull/25#discussion_r1010616094)

Reply:
If we want to get rid of the fallback libraries however we might need to either make winston-js browser friendly (e.g. see https://github.com/winstonjs/winston/issues/2196) or from our end (sdk-js) make the logging different for browser vs node.

Reply:
This https://github.com/winstonjs/winston/issues/287 was opened in 2013

It's very likely this will take us month(s) to get browser support in winston,
I'd recommend you try this: https://stackoverflow.com/a/55433049/843194
and else, "just" return a `console` object when we're in the browser.

We can tackle full support for logging as a follow-up issue (related: https://github.com/testground/testground/issues/1355)

Reply:
Not to worry, that’s already an approach I’m taking on my local replace branch. Sorry if i didn’t communicate that yet